lib/weka/core/attribute.rb

```diff
@@ -70,6 +76,17 @@ def internal_value_of(value)
         return parse_date(value.to_s)     if date?
         return value.to_f                 if numeric?
         return index_of_value(value.to_s) if nominal?
+
+        if string?
+          # string has unlimited range of possible values. We check if
+          # it has been added. If yes then we simply return the index
+          if (idx = index_of_value(value.to_s)) != -1
+            idx
+          else
+            # else we add the value to the range and return its new index
+            add_string_value(value.to_s)
+          end
+        end
```

Comment:
I think the string handling could be moved into a separate private method, in order to keep `#internal_value_of` as concise as possible. Maybe sth. like `index_of_string(value)`?

With this, we could have a one-liner as for the other types (I'd use the `return` here as well, even if it is not necessary):

```ruby
return index_of_string(value) if string?
```

I would also avoid assignments in if clauses and would rather do it before with a fully readable name, like:

```ruby
index = index_of_value(value.to_s)

if index == -1
  # ...
else
 # ...
end
```

The comments could then be the method’s comment and move out of the body (Should work, right? The new method should be short enough then, to get what it refers to)

Could you make these changes? Then we are good to go!